ocaml/xapi/create_misc.ml

```diff
@@ -573,3 +573,8 @@ let create_chipset_info ~__context =
 	let info = ["iommu", iommu] in
 	Db.Host.set_chipset_info ~__context ~self:host ~value:info
 
+let create_patches_requiring_reboot_info ~__context ~host =
+  let patch_uuids = Stdext.Unixext.read_lines !Xapi_globs.reboot_required_hfxs in
+  let host_patches = Db.Host.get_patches ~__context ~self:host in
+  let pool_patches = List.map (fun hp -> Db.Host_patch.get_pool_patch ~__context ~self:hp) host_patches in
+  List.filter (fun patch -> List.mem (Db.Pool_patch.get_uuid ~__context ~self:patch) patch_uuids) pool_patches
```

Comment:
I don't think we need to go through the `host_patches` here? Can't we just rely on the UUIDs in the file to be right for the host (since the file is on the local host)? Then the function becomes quite simple (doing the DB set here as well):
```
let patch_uuids = Stdext.Unixext.read_lines !Xapi_globs.reboot_required_hfxs in
let patches = List.map (fun uuid -> Db.Pool_patch.get_by_uuid ~__context ~uuid) patch_uuids in
Db.Host.set_patches_requiring_reboot ~__context ~self:host ~value:patches
```